ArduPlane/tiltrotor.cpp

```diff
@@ -310,24 +341,89 @@ void Tiltrotor::continuous_update(void)
+                    if (_transition_fwd_tilt_frac < deg2quad((float)max_angle_deg)) {
+                        // use the normal rate up to the maximum VTOL tilt angle
+                        if (max_rate_down_dps <= 0) {
+                            // use the up rate if the down rate isn't defined
+                            tilt_rate_dps = max_rate_up_dps;
+                        } else {
+                            tilt_rate_dps = max_rate_down_dps;
+                        }
+                    } else {
+                        // when tilting past the maximum VTOL tilt angle set by  Q_TILT_MAX,
+                        // enable use of a different rate which can be set slower to allow the
+                        // pitch angle and height control time to raise throttle on the tilted motors.
+                        if (max_rate_down_transition_dps > 0) {
+                            tilt_rate_dps = max_rate_down_transition_dps;
+                        } else if (max_rate_down_dps > 0) {
+                            // use the VTOL flight tilt down rate if the rate to be used when past Q_TILT_MAX isn't set
+                            tilt_rate_dps = max_rate_down_dps;
+                        } else {
+                            // use the VTOL flight tilt up rate if the VTOL flight tilt down rate isn't set
+                            tilt_rate_dps = max_rate_up_dps;
+                        }
+                    }
+                    new_tilt = get_forward_flight_tilt();
+                } else if (_transition_fwd_tilt_frac > deg2quad((float)max_angle_deg)) {
+                    // If the maximum throttle of the tilting motors is clipping and rotors are tilted past the normal
+                    // VTOL limit, tilt the rotors back faster to prevent possible loss of attitude control.
+                    if (max_rate_down_transition_dps > 0) {
+                        tilt_rate_dps = max_rate_down_transition_dps;
+                    } else if (max_rate_down_dps > 0) {
+                        // use the VTOL flight tilt down rate if the rate to be used when past Q_TILT_MAX isn't set
+                        tilt_rate_dps = max_rate_down_dps;
+                    } else {
+                        // use the VTOL flight tilt up rate if the VTOL flight tilt down rate isn't set
+                        tilt_rate_dps = max_rate_up_dps;
+                    }
+                    tilt_rate_dps = (3 * tilt_rate_dps) / 2;
+                    new_tilt = deg2quad((float)max_angle_deg);
+
+                }
+
+                AP::logger().Write("QTLT", "TimeUS,MT,NT,TFTF", "Qfff",
+                                        AP_HAL::micros64(),
+                                        (double)max_throttle,
+                                        (double)new_tilt,
+                                        (double)_transition_fwd_tilt_frac);
```

Comment:
This has not been updated yet at this point, so its from the last loop still.

Reply:
That's OK